src/Raven.Server/Documents/Handlers/AI/Agents/AiAgentProcessorForGetAiAgent.cs

```diff
@@ -20,20 +22,21 @@ public AiAgentProcessorForGetAiAgent([NotNull] TRequestHandler requestHandler) :
     public override async ValueTask ExecuteAsync()
     {
         using var token = RequestHandler.CreateHttpRequestBoundOperationToken();
-        var name = RequestHandler.GetStringQueryString("name");
+        var identifier = RequestHandler.GetStringQueryString("identifier");
 
-        Dictionary<string, AiAgentConfiguration> agents;
+        List<AiAgentConfiguration> agents;
         using (ServerStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext ctx))
         using (ctx.OpenReadTransaction())
         using (var record = ServerStore.Cluster.ReadRawDatabaseRecord(ctx, RequestHandler.DatabaseName))
         {
             agents = record.AiAgents;
         }
 
-        if (string.IsNullOrEmpty(name) == false)
+        if (string.IsNullOrEmpty(identifier) == false)
         {
-            if (agents.TryGetValue(name, out var configuration) == false)
-                throw new ArgumentException($"AI Agent '{name}' doesn't exists");
+            var configuration = agents.FirstOrDefault(c => c.Identifier == identifier);
```

Comment:
are those identifiers case sensitive? They should be not IMO?

Reply:
also in RawDatabaseRecord you have a method TryGetAgent, using it would guarantee you having same search logic in each place